include/camp/resource.hpp

```diff
@@ -94,42 +94,86 @@ namespace resources
       void wait() { m_value->wait(); }
 
       /*
-       * \brief Compares two Resources to see if they are equal. Two Resources are equal if they are
-       * on the same platform and they describe the same stream (i.e. CUDA, HIP) or queue (i.e. SYCL).
-       * (Note: This operator overload is on the generic Resource. Using the ContextModel's operator
-       * overload, we can call the specific resource's (i.e. T) operator== to compare streams or
-       * queues.)
+       * \brief Compares two Resources to see if they are equal. Two Resources are equal if they
+       * have the same ID (same platform and same stream/queue).
        *
-       * \return True or false depending on comparison with m_value if they are on the same platform.
+       * \return True if they have the same ID, false otherwise.
        */
       bool operator==(Resource const& r) const
       {
-        if(get_platform() == r.get_platform()) {
-          return (m_value->compare(r));
-        }
-        return false;
+        return (get_id() == r.get_id());
       }
+
       /*
-       * \brief Compares two Resources to see if they are NOT equal. Two Resources are not equal
-       * if they are on separate platforms. Also, even if they are on the same platform, if they
-       * describe different streams (i.e. CUDA, HIP) or different queues (i.e. SYCL), then they 
-       * are not equal.
+       * \brief Compares two Resources to see if they are NOT equal.
        *
-       * \return Negation of == operator. 
+       * \return Negation of == operator.
        */
       bool operator!=(Resource const& r) const
       {
         return !(*this == r);
       }
 
+      /*
+       * \return True if this resource's ID is less than the other's.
+       */
+      bool operator<(Resource const& r) const
+      {
+        return (get_id() < r.get_id());
```

Comment:
If we do this it would be better to do a lexicographical comparison of platform and then their underlying values to avoid cases where the hash/id is potentially not unique. We could probably extend compare to be like the c++20 spaceship operator and return less, equal, or greater.

Reply:
Actually it might be better to rename get_id to hash because its possible, unlikely, but possible that two could end up with the same key.

Reply:
Since I got rid of operators <, >, <=, >=, does this still apply? I could add to the operator== an extra check like:
```
102       bool operator==(Resource const& r) const
103       {
104         if (get_platform() != r.get_platform()) return false;
105         return (get_id() == r.get_id());
106       }
```
if that would be sufficient

Reply:
I don't think I understand what you're getting at here